lib/elasticity/base_document.rb

```diff
@@ -49,5 +42,41 @@ def delete
     def created?
       @created || false
     end
+
+    def self.elasticity_attributes(*args)
```

Comment:
Maybe with a nomenclature change this could work. Nothing about this code is actually specific to active record. The question is does the external ruby object define public accessors for all the attributes defined in the document? If so then this works.

`from_external_source` maybe?